federationapi/routing/backfill.go

```diff
@@ -105,6 +110,12 @@ func Backfill(
 		eventJSONs = append(eventJSONs, e.JSON())
 	}
 
+	// sytest wants these in reversed order, similar to /messages, so reverse them now.
+	for i := len(eventJSONs)/2 - 1; i >= 0; i-- {
```

Comment:
This feels a bit strange to do here - should the API just process `QueryBackfill` in reverse order?

Reply:
The thing which demands it being in that order is the `/backfill` endpoint, not `QueryBackfill`, so I want to keep the code which is making it conform to the API as close as possible to the endpoint which demands it, if that makes sense.